### PR TITLE
Housecleaning

### PR DESCRIPTION
**Pull request by jensens**
- pyupgrade
- isort
- black
- remove six
- move imports from CMFPlone to plone.base
- copy `munge_search_terms` from CMFPlone to here to avoid dependency on CMFPlone.

### CODE REVIEW

**Comment by gforcada** [comment]
Thanks for adding this, though, if we don't do it via `plone/meta` the next contribution might break pyupgrade/isort/black/etc and no one will notice, getting worse and worse by each contribution and finally having to create yet another clean up PR 😵‍💫 

Let's merge this and I will create a follow up PR with the missing bits 👍🏾

File: plone/app/querystring/querybuilder.py [hunk 1]
```diff
@@ -18,10 +16,56 @@
...
+
+
+def _quote(term):
+    # The terms and, or and not must be wrapped in quotes to avoid
+    # being parsed as logical query atoms.
+    if term.lower() in ('and', 'or', 'not'):
+        term = '"%s"' % term
+    return term
+
+def munge_search_term(query):
```
Comment (gforcada) [inline plone/app/querystring/querybuilder.py:48]: A quick search shows that it is only used here, on `CMFPlone/browser/search.py` and its related tests. Do you plan to make CMFPlone import this and move also the tests here?

Should this rather not go into `plone.base` instead, though? I see potentially other distributions (collective.solr, collections (`plone.app.content` or `contenttypes`, `plone.app.portlets` ?) that might reuse this...

As a first step to untangle the dependency is fine, but maybe we can do this move to `plone.base`?

File: plone/app/querystring/tests/registry_testdata.py [hunk 2]
```diff
@@ -1,66 +1,61 @@
...
-                            'operation': u'plone.app.querystring.queryparser.'
-                                         u'_equal',
-                            'description': u'Tip: you can use * to '
-                                           u'autocomplete.',
-                            'title': u'equals'
+                    "string": {
+                        "is": {
+                            "widget": None,
+                            "operation": "plone.app.querystring.queryparser." "_equal",
+                            "description": "Tip: you can use * to " "autocomplete.",
```
Comment (gforcada) [inline plone/app/querystring/tests/registry_testdata.py:54]: That's a bit annoying, but `black` wants to preserve the AST as a way to verify that it did not break anything, but then it leaves these strings that could be join together 
```suggestion
                            "operation": "plone.app.querystring.queryparser._equal",
                            "description": "Tip: you can use * to autocomplete.",
```

File: pyproject.toml [hunk 3]
```diff
@@ -18,3 +18,10 @@ showcontent = true
 directory = "bugfix"
 name = "Bug fixes:"
 showcontent = true
+
+[tool.isort]
```
Comment (gforcada) [inline pyproject.toml:22]: that's wrong, see [plone/meta configuration](https://github.com/plone/meta/blob/master/config/default/pyproject.toml.j2#L37-L38)

File: plone/app/querystring/querybuilder.py (hunk 1 above)
Comment (jensens) [reply plone/app/querystring/querybuilder.py:48]: it is used here and in CMFPlone. So I do not expect it to be used somewhere else. I would keep it here and later on move tests here and import in CMFPlone frim here.

File: plone/app/querystring/tests/testQueryParser.py [hunk 4]
```diff
@@ -127,398 +138,345 @@ def getAuthenticatedMember(self):
...
+            "o": "plone.app.querystring.operation.string.is",
+            "v": "Welcome to Plone",
         }
-        parsed = queryparser.parseFormquery(MockSite(), [data, ])
-        self.assertEqual(parsed, {'Title': {'query': 'Welcome to Plone'}})
+        parsed = queryparser.parseFormquery(
+            MockSite(),
+            [
+                data,
+            ],
```
Comment (gforcada) [inline plone/app/querystring/tests/testQueryParser.py:174]:
```suggestion
                data
            ]
```

with the above and running `black` again, you will get it back to a single line 😄

File: plone/app/querystring/tests/testQueryParser.py [hunk 5]
```diff
@@ -127,398 +138,345 @@ def getAuthenticatedMember(self):
...
+            "v": "Welcome to Plone",
         }
         parsed = queryparser.parseFormquery(
-            MockSite(), [data, ],
-            sort_on='sortable_title',
-            sort_order='reverse')
+            MockSite(),
+            [
+                data,
+            ],
```
Comment (gforcada) [inline plone/app/querystring/tests/testQueryParser.py:188]:
```suggestion
                data
            ]
```

File: plone/app/querystring/querybuilder.py (hunk 1 above)
Comment (jensens) [reply plone/app/querystring/querybuilder.py:48]: If this is OK I will create a ticket in CMFPlone so we don't forget that

File: plone/app/querystring/tests/testQueryParser.py [hunk 6]
```diff
@@ -127,398 +138,345 @@ def getAuthenticatedMember(self):
...
         parsed = queryparser.parseFormquery(
-            MockSite(), [data, ],
-            sort_on='unknown',
-            sort_order='reverse')
-        self.assertEqual(
-            parsed, {'Title': {'query': 'Welcome to Plone'}})
+            MockSite(),
+            [
+                data,
+            ],
```
Comment (gforcada) [inline plone/app/querystring/tests/testQueryParser.py:211]: _ditto_ 🤓

File: plone/app/querystring/tests/testQueryParser.py [hunk 7]
```diff
@@ -127,398 +138,345 @@ def getAuthenticatedMember(self):
...
+            "v": "/foo",
         }
-        parsed = queryparser.parseFormquery(MockSite(), [data, ])
-        self.assertEqual(
-            parsed, {'path': {'query': ['/%s/foo' % MOCK_SITE_ID]}})
+        parsed = queryparser.parseFormquery(
+            MockSite(),
+            [
+                data,
+            ],
```
Comment (gforcada) [inline plone/app/querystring/tests/testQueryParser.py:227]: _ditto_

File: plone/app/querystring/tests/testQueryParser.py [hunk 8]
```diff
@@ -127,398 +138,345 @@ def getAuthenticatedMember(self):
...
+            "v": "00000000000000001",
         }
-        parsed = queryparser.parseFormquery(MockSite(), [data, ])
-        self.assertEqual(
-            parsed, {'path': {'query': ['/%s/foo' % MOCK_SITE_ID]}})
+        parsed = queryparser.parseFormquery(
+            MockSite(),
+            [
+                data,
+            ],
```
Comment (gforcada) [inline plone/app/querystring/tests/testQueryParser.py:241]: _ditto_

there are quite a few others, it's pointless to keep adding comments for each of them 😅

File: plone/app/querystring/queryparser.py [hunk 9]
```diff
@@ -60,31 +58,36 @@ def parseFormquery(context, formquery, sort_on=None, sort_order=None):
...
 def _excludes(context, row):
-    return {row.index: {'not': row.values}}
+    return {row.index: {"not": row.values}}
 
 
 def _equal(context, row):
-    return {row.index: {'query': row.values, }}
+    return {
+        row.index: {
+            "query": row.values,
```
Comment (gforcada) [inline plone/app/querystring/queryparser.py:84]:
```suggestion
            "query": row.values
```

File: plone/app/querystring/queryparser.py [hunk 10]
```diff
@@ -99,39 +102,51 @@ def _intEqual(context, row):
             values = int(row.values)
         except (ValueError, TypeError, AttributeError):
             pass
-    return {row.index: {'query': values, }}
+    return {
+        row.index: {
+            "query": values,
```
Comment (gforcada) [inline plone/app/querystring/queryparser.py:107]:
```suggestion
            "query": values
```

**Review by gforcada — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 3 above)
Comment (jensens) [reply pyproject.toml:22]: well, essentially the same, but good to know.

**Comment by gforcada** [comment]
@jenkins-plone-org please run jobs

**Review by gforcada — APPROVED** [review]

**Pull request merged**